R/version.R

```diff
@@ -185,12 +185,19 @@ format.version_sentinel <-
     function()
 {
     bioc <- tryCatch({
-        packageVersion("BiocVersion")[, 1:2]
-    }, error = identity)
-    if (inherits(bioc, "error"))
+        .get_BiocVersion_version()
+    },
+        error = function(e) {
+            .version_sentinel(conditionMessage(e))
+        },
+        warning = function(w) {
+            .version_sentinel(conditionMessage(w))
```

Comment:
usually, it is wrong to use `tryCatch()` to a catch a warning, because code after the warning is not evaluated... use `withCallingHandlers()` instead. E.g., here we're expecting the warning to be handled but code evaluation to continue and return `1`...

```
> tryCatch({ warning("oops"); 1 }, warning = function(w) {})
NULL
> withCallingHandlers({ warning("oops"); 1 }, warning = function(w) { invokeRestart("muffleWarning")})
[1] 1
```
so to handle both I think we're at
```
    bioc <-  withCallingHandlers({
        tryCatch({
            get_BiocVersion_version()
        }, error = function(e) {
            ## an error if 'BiocVersion' not found (R < 4.3)...
            .version_sentinel(conditionMessage(e))
        }), warning = function(w) {
            ## ... or warning and 'NA' (R >= 4.3)
            invokeRestart("muffleWarning")
        }
    })
```